### PR TITLE
add order history.

### PR DESCRIPTION
**Pull request by Alice1319**
fixes https://github.com/eclipse-xpanse/xpanse/pull/2030

Snipaste_2024-12-03_16-02-27

Snipaste_2024-12-03_111

Snipaste_2024-12-03_16-00-50

Snipaste_2024-12-03_16-02-01

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
LGTM

**Comment by profoundwu** [comment]
LGTM

**Comment by luoping-12345** [comment]
LGTM

**Comment by baixinsui** [comment]
LGTM

**Comment by dengyali0125** [comment]
LGTM

File: src/components/content/deployedServices/myServices/MyServiceHistory.tsx [hunk 1]
```diff
@@ -24,20 +27,80 @@ export const MyServiceHistory = ({
...
+            filters: taskTypeFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: React.Key | boolean, record) => record.taskType.startsWith(value.toString()),
+            render: (value: taskType) => {
+                return <div className={serviceModifyStyles.serviceHistoryValue}>{value}</div>;
+            },
         },
         {
             title: 'Previous',
```
Comment (swaroopar) [inline src/components/content/deployedServices/myServices/MyServiceHistory.tsx:106]: I would suggest, that we remove the previous and new column and replace it with details column.. and in that on mouse over, we can display all this information.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: src/components/content/deployedServices/myServices/MyServiceHistory.tsx [hunk 2]
```diff
@@ -128,15 +214,21 @@ export const MyServiceHistory = ({
...
-                        <Tooltip title={value}>
-                            <span className={serviceModifyStyles.modifyHistoryErrorMsgValue}>{value}</span>
-                        </Tooltip>
+                        <Popover
+                            content={<pre className={serviceModifyStyles.serviceOrderErrorText}>{value.details}</pre>}
+                            title={value.errorType}
+                            trigger='hover'
+                        >
+                            <Button className={serviceModifyStyles.serviceOrderErrorDataHover} type={'link'}>
+                                {'error response'}
```
[the remark this answers is not in the document]
Comment (Alice1319) [reply src/components/content/deployedServices/myServices/MyServiceHistory.tsx:229]: fixed.

File: src/components/content/deployedServices/myServices/MyServiceHistory.tsx [hunk 3]
```diff
@@ -24,20 +27,80 @@ export const MyServiceHistory = ({
...
+            };
+            filters.push(filter);
+        });
+        taskStatusFilters = filters;
     }
 
     const columns: ColumnsType<ServiceOrderDetails> = [
         {
-            title: 'ModifyId',
+            title: 'OrderId',
```
[the remark this answers is not in the document]
Comment (Alice1319) [reply src/components/content/deployedServices/myServices/MyServiceHistory.tsx:81]: fixes like this:
image

image

File: src/components/content/deployedServices/myServices/MyServiceHistory.tsx (hunk 1 above)
Comment (Alice1319) [reply src/components/content/deployedServices/myServices/MyServiceHistory.tsx:106]: fixes like this:
image

image

File: src/components/content/deployedServices/myServices/MyServiceHistory.tsx [hunk 4]
```diff
@@ -92,22 +136,45 @@ export const MyServiceHistory = ({
...
+            title: 'TaskStatus',
             dataIndex: 'taskStatus',
             align: 'center',
             width: 50,
+            filters: taskStatusFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: React.Key | boolean, record) => record.taskStatus.startsWith(value.toString()),
             render: (value) => {
                 if (value === taskStatus.FAILED) {
```
Comment (swaroopar) [inline src/components/content/deployedServices/myServices/MyServiceHistory.tsx:175]: we need some better handling for other two task states
CREATED("created"), - can be in grey color with a corresponding icon
IN_PROGRESS("in-progress"), - orange and loading icon

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: src/components/content/deployedServices/myServices/MyServiceHistory.tsx (hunk 4 above)
Comment (Alice1319) [reply src/components/content/deployedServices/myServices/MyServiceHistory.tsx:175]: fixes like this:
image

image

**Comment by WangLiNaruto** [comment]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**